waitress/task.py

```diff
@@ -203,7 +211,12 @@ def build_response_header(self):
                 [x.capitalize() for x in headername.split('-')]
             )
             if headername == 'Content-Length':
-                content_length_header = headerval
+                if self.has_body:
+                    content_length_header = headerval
+                else:
+                    del response_headers[i]
+                    continue  # pragma: no cover
```

Comment:
deletion while iteration - sure this is ok?

Reply:
We do the same thing here: https://github.com/Pylons/waitress/pull/202/files/f9ad8bbb56aab91dd2847e4137b0e94bea1791c8#diff-6965458a8fd3a611b9b9d416d9cdff67L277

and that code has been around for years.

Reply:
I guess I need to find out if enumerate creates a copy of the list or if it iterates over the list internally.

Reply:
This will break in subtle ways if there is more than one Content-Length header set (which is illegal, but possible)

Reply:
FWIW, enumerate has generator semantics (see PEP 279).

Reply:
Then the other code and this code are both subtly broken... will fix